internal/ui/src/hooks/use-keyboard-shortcut.tsx

```diff
@@ -0,0 +1,299 @@
+      if (part.length > 0) {
+        if (keyAssigned) {
+          console.warn(
+            `[useKeyboardShortcut] Multiple non-modifier keys detected in shortcut: "${shortcut}"`,
+          );
+          return null;
+        }
+        combo.key = part; // Store original key name
+        combo.code = getKeyNameToCode(part); // Attempt to get the code
+        if (!combo.code) {
+          console.warn(
+            `[useKeyboardShortcut] Failed to map key "${part}" to code for shortcut: "${shortcut}". Please check spelling or expand getKeyNameToCode mapping.`,
+          );
+          return null; // Fail parsing if code cannot be determined
+        }
+        keyAssigned = true;
+      } else {
+        console.warn(`[useKeyboardShortcut] Empty part detected in shortcut string: "${shortcut}"`);
+        return null;
+      }
+    }
+  }
+
+  // Final validation: Ensure a key and code were actually assigned
+  if (!keyAssigned || !combo.key || !combo.code) {
+    console.warn(`[useKeyboardShortcut] No valid key/code identified for shortcut: "${shortcut}"`);
+    return null;
+  }
+
+  // Return the full KeyCombo object with explicit boolean values for modifiers
+  return {
+    key: combo.key,
+    code: combo.code,
+    ctrl: !!combo.ctrl,
+    meta: !!combo.meta,
+    shift: !!combo.shift,
+    alt: !!combo.alt,
+  };
+};
+
+/**
+ * A React hook to trigger a callback function when a specific keyboard shortcut is pressed globally.
+ * Matches shortcuts based on KeyboardEvent.code (physical key) for better reliability, especially with Alt/Option keys.
+ * Handles complex shortcut strings (e.g., "ctrl+shift+k") or pre-defined KeyCombo objects.
+ * Includes options to ignore inputs, contentEditable elements, prevent default behavior, and disable the listener.
+ *
+ * @param shortcut The keyboard shortcut definition (string like "ctrl+alt+e", "shift+enter", or KeyCombo object). Can be null/undefined to disable. Key names should be standard (e.g., 'a', 'enter', 'f1', 'comma').
+ * @param callback The function to execute when the shortcut is matched. Should be memoized (e.g., with useCallback) if defined inline or stable otherwise.
+ * @param options Optional configuration for the shortcut listener (KeyboardShortcutOptions). Options are merged with defaults. Provide a stable reference (e.g., useMemo) if passing an object.
+ */
+export function useKeyboardShortcut(
+  shortcut: string | KeyCombo | null | undefined,
+  callback: () => void,
+  options?: KeyboardShortcutOptions,
+): void {
+  const mergedOptions = { ...defaultOptions, ...options };
+
+  const { preventDefault, ignoreInputs, ignoreContentEditable, disabled } = mergedOptions;
+
+  // Memoize callback for stability in the useEffect dependency array
+  // biome-ignore lint/correctness/useExhaustiveDependencies: callback is intentionally excluded to prevent infinite re-renders
+  const memoizedCallback = useCallback(callback, [callback]);
```

Comment:
your linter comment says its excluded, but you've included the callback in the dependency array.